src/status_im/utils/ethereum/eip55.cljs

```diff
@@ -5,18 +5,17 @@
 
    e.g. 0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed"
   (:require [clojure.string :as string]
-            [status-im.utils.ethereum.core :as ethereum]))
+            [status-im.js-dependencies :as dependencies]))
 
-(defn valid-address-checksum? [address]
+(defn address->checksum
+  "Converts an arbitrary case address to one with correct checksum case."
+  [address]
+  (when address
+    (.toChecksumAddress dependencies/Web3.prototype address)))
+
+(defn valid-address-checksum?
+  "Checks address checksum validity, i.e. if the address had the correct checksum case to begin with,
+   the result will be the same as input"
+  [address]
   "verify address checksum according to EIP 55"
-  (let [adHash (ethereum/naked-address (ethereum/sha3
-                                        (string/lower-case (ethereum/naked-address address))))]
-    (every? true?
-            (map-indexed (fn [idx char]
-                           (if (> (compare char "9") 0)
-                             (if (>= (js/parseInt (nth adHash idx) 16) 8)
-                                ;  If true should be upper case
-                               (<= (compare char "Z") 0)
-                                ; If not should be lower case
-                               (> (compare char "Z") 0))
-                             true)) (ethereum/naked-address address)))))
\ No newline at end of file
+  (= address (address->checksum address)))
```

Comment:
I actually like the old one as it was one less dep on web3.

Reply:
that's true, but then again, I like the new one as it's less code and therefore less opportunities for bugs

Reply:
we want to drop web3, can we use https://web3js.readthedocs.io/en/1.0/web3-utils.html#tochecksumaddress here ?

Reply:
that's ok with me, I'll use that dep then

Reply:
ok, it's web2-utils now

Reply:
web2 :(

Reply:
haha, just a typo, no judgment expressed :P